File: backend/services/post-rust-actix-web-fdm/src/common/api/invalid_url_post_content.rs
use crate::common::utils::error::*;

impl super::super::UrlPostContent {
    pub(in crate::common) fn url_post_content_empty() -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "URL_POST_CONTENT_EMPTY".into(),
                reason: "url is empty".to_string(),
                message: "url is empty".to_string(),
            },
        }
    }
    pub(in crate::common) fn url_post_content_too_long() -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "URL_POST_CONTENT_TOO_LONG".into(),
                reason: "url is too long".to_string(),
                message: "url is too long".to_string(),
            },
        }
    }
    pub(in crate::common) fn url_post_content_invalid(err: url::ParseError) -> ErrorBody {
        ErrorBody {
            error: ErrBody {
                error: "URL_POST_CONTENT_INVALID".into(),
                reason: format!("{}", err),
                message: format!("url is invalid: {}", err),
            },
        }
    }
}

File: backend/services/post-kt-vertx-fdm/src/main/kotlin/me/jason5lee/post_kt_vertx_fdm/common/api/InvalidUrlPostContent.kt
package me.jason5lee.post_kt_vertx_fdm.common.api

import me.jason5lee.post_ktor_mongo_fdm.common.utils.Err
import me.jason5lee.post_ktor_mongo_fdm.common.utils.FailureBody

internal object InvalidUrlPostContent {
    internal val empty: FailureBody = FailureBody(
        error = Err(
            error = "URL_POST_CONTENT_EMPTY",
            reason = "URL post content cannot be empty",
            message = "URL post content cannot be empty",
        )
    )
    internal val tooLong: FailureBody = FailureBody(
        error = Err(
            error = "URL_POST_CONTENT_TOO_LONG",
            reason = "URL post content must be at most 65535 characters",
            message = "URL post content must be at most 65535 characters",
        )
    )

    fun invalid(reason: String?): FailureBody = FailureBody(
        error = Err(
            error = "URL_POST_CONTENT_INVALID",
            reason = reason ?: "URL post content must be a valid URL",
            message = "URL post content must be a valid URL",
        )
    )
}
